compiler/src/iree/compiler/Dialect/LinalgExt/IR/TilingInterfaceImpl.cpp

```
@@ -1440,6 +1441,220 @@ LogicalResult ArgCompareOp::generateScalarImplementation(OpBuilder &b,
   return success();
 }
 
+FailureOr<SmallVector<Value>>
+ArgCompareOp::generateInitialTensorForPartialReduction(
+    OpBuilder &b, Location loc, ArrayRef<OpFoldResult> sizes,
+    const llvm::SetVector<unsigned> &reductionDims) {
+  // Get the original init tensors.
+  Value valueInit = outputValue();
+  Value indexInit = outputIndex();
+
+  // Create tensors with the partial result shape.
+  Type valueElTy = getElementTypeOrSelf(valueInit.getType());
+  Type indexElTy = getElementTypeOrSelf(indexInit.getType());
+  SmallVector<OpFoldResult> partialResultShape(sizes);
+  Value emptyValueTensor =
+      tensor::EmptyOp::create(b, loc, partialResultShape, valueElTy);
+  Value emptyIndexTensor =
+      tensor::EmptyOp::create(b, loc, partialResultShape, indexElTy);
+
+  // Broadcast init values to partial result shape for slicing inside
+  // scf.forall. Each tile in the parallel loop will extract slices from
+  // these broadcasted tensors to get initialized values for the ArgCompareOp.
+  // Example: tensor<64xf32> -> tensor<64x32xf32> for 32 reduction tiles along
+  // dim 1.
+  auto broadcastDims = llvm::to_vector_of<int64_t>(reductionDims);
+  Operation *valueBroadcastOp = linalg::BroadcastOp::create(
+      b, loc, valueInit, emptyValueTensor, broadcastDims);
+  Operation *indexBroadcastOp = linalg::BroadcastOp::create(
+      b, loc, indexInit, emptyIndexTensor, broadcastDims);
+
+  return SmallVector<Value>{valueBroadcastOp->getResult(0),
+                            indexBroadcastOp->getResult(0)};
+}
+
+FailureOr<TilingResult> ArgCompareOp::tileToPartialReduction(
+    OpBuilder &b, Location loc, ReductionTilingStrategy strategy,
+    ValueRange init, ArrayRef<OpFoldResult> offsets,
+    ArrayRef<OpFoldResult> sizes,
+    const llvm::SetVector<unsigned> &reductionDims,
+    ArrayRef<OpFoldResult> splitReductionIvs) {
+  assert(strategy == ReductionTilingStrategy::PartialReductionOuterParallel &&
+         "Requires PartialReductionOuterParallel strategy");
```

Just curious, is there a reason why this is an assert instead of returning failure?

---

Thanks, it is better to return failure.